src/contrib/cluster/Akka.Cluster.Sharding.Tests.MultiNode/Delivery/ClusterShardingDeliveryGracefulShutdownSpec.cs

```diff
@@ -276,12 +276,12 @@ private void Cluster_sharding_must_gracefully_shutdown_the_oldest_region()
                     var producer = CreateProducer("p-2");
                     AwaitAssert(() =>
                     {
-                        var responses = Enumerable.Range(1, 20).Select(n =>
+                        var maxCount = 20;
+                        foreach(var n in Enumerable.Range(1, maxCount))
                         {
                             producer.Tell(n, TestActor);
-                            return ExpectMsg(n, TimeSpan.FromSeconds(1));
-                        }).ToImmutableHashSet();
-
+                        }
+                        var responses = ReceiveN(maxCount, TimeSpan.FromSeconds(2));
```

Comment:
Relaxed the assertion here since it's possible that messages can get re-queued out of order / partially delivered from last time in the event of a timeout, consider that they're all going back to the same actor again